include/veriblock/blockchain/base_block_tree.hpp

```diff
@@ -239,9 +299,11 @@ struct BaseBlockTree {
   }
 
   bool operator==(const BaseBlockTree& o) const {
-    TreeFieldsComparator cmp{};
-    return cmp(blocks_, o.blocks_) && cmp(tips_, o.tips_) &&
-           (activeChain_ == o.activeChain_);
+    CollectionOfPtrComparator cmp{};
+    bool a = cmp(blocks_, o.blocks_);
+    bool b = cmp(tips_, o.tips_);
+    bool c = (activeChain_ == o.activeChain_);
+    return a && b && c;
```

Comment:
We might avoid unnecessary comparisons using the previous code.

Reply:
compiler will optimize this away, but at the same time this layout simplifies debugging